File: lab4/hibernate_lab4/src/main/java/org/example/Invoice.java
package org.example;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int InvoiceID;
    private int InvoiceNumber;
    private int Quantity;
    @ManyToMany(cascade = {CascadeType.PERSIST})
    Set<Product> Products = new HashSet<>();

    public Invoice(){}

    public Invoice(int invoiceNumber, int quantity){
        this.InvoiceNumber = invoiceNumber;
        this.Quantity = quantity;
    }

    public void addProduct(Product product){
        this.Products.add(product);
    }

    public Set<Product> getProducts(){
        return this.Products;
    }
    public String toString(){
        return Integer.toString(this.InvoiceNumber);
    }
}

File: lab3/KacperCwiertniaEFProducts/KacperCwiertniaEFProducts/Invoice.cs
using System;
namespace KacperCwiertniaEFProducts
{
	public class Invoice
	{
        public Invoice(){
            this.Products = new HashSet<Product>();
        }

        public int InvoiceID { get; set; }
        public int InvoiceNumber { get; set; }
        public int Quantity { get; set; }
        public virtual ICollection<Product> Products { get; set; }
	}
}

